jackson-apt-processor/src/main/java/org/dominokit/jacksonapt/processor/AbstractCollectionMapperGenerator.java

```diff
@@ -0,0 +1,81 @@
+package org.dominokit.jacksonapt.processor;
+
+import static org.dominokit.jacksonapt.processor.AbstractMapperProcessor.elementUtils;
+import static org.dominokit.jacksonapt.processor.AbstractMapperProcessor.typeUtils;
+import static org.dominokit.jacksonapt.processor.ObjectMapperProcessor.DEFAULT_WILDCARD;
+
+import java.util.List;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.Name;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeMirror;
+
+import org.dominokit.jacksonapt.JsonDeserializer;
+import org.dominokit.jacksonapt.JsonSerializer;
+import org.dominokit.jacksonapt.processor.deserialization.FieldDeserializersChainBuilder;
+import org.dominokit.jacksonapt.processor.serialization.FieldSerializerChainBuilder;
+
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.CodeBlock;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+
+
+public abstract class AbstractCollectionMapperGenerator extends AbstractMapperGenerator {
+	protected MethodSpec makeNewDeserializerMethod(Element element, Name beanName) {
+		return MethodSpec.methodBuilder("newDeserializer")
+				.addModifiers(Modifier.PROTECTED)
+				.addAnnotation(Override.class)
+				.returns(ParameterizedTypeName.get(ClassName.get(JsonDeserializer.class),
+						ClassName.get(getElementType(element))))
+				.addCode(
+						CodeBlock.builder()
+						.add("return ")
+						.add(new FieldDeserializersChainBuilder(getElementType(element)).getInstance(getElementType(element)))
```

Comment:
@niloc132 I got your point but is not an easy fix. What we need here is a notion for an expression i.e. "return [expression]", but it seems the javapoet does not have such metaphor. 
Probably it is better to include such restriction into the contract for:
`FieldDeserializersChainBuilder.getInstance(..)`
and
`FieldSerializersChainBuilder.getInstance(..)`

i.e. the CodeBlock returned from getInstance() must be valid java expression.

Reply:
Agreed, and as you noted elsewhere, the original code was also making this mistake.

Valid Java expression is one option, along the same lines you could just make it a plain String (since this is what add(CodeBlock) will do anyway, this way without ambiguity of it _being_ a code block), a third could be to actually make it return a valid Java codeblock - let it generate the `return` and `;` too?

I'm leaning toward the last, but perhaps in another patch, it would be a bigger change.